src/OnlineSales/Services/ContactService.cs

```diff
@@ -0,0 +1,70 @@
+﻿// <copyright file="ContactService.cs" company="WavePoint Co. Ltd.">
+// Licensed under the MIT license. See LICENSE file in the samples root for full license information.
+// </copyright>
+
+using OnlineSales.Data;
+using OnlineSales.Entities;
+using OnlineSales.Interfaces;
+
+namespace OnlineSales.Services
+{
+    public class ContactService : IContactService
+    {
+        private readonly ApiDbContext apiDbContext;
+
+        public ContactService(ApiDbContext apiDbContext)
+        {
+            this.apiDbContext = apiDbContext;
+        }
+
+        public async Task<Contact> AddContact(Contact contact)
+        {
+            var contactWrappWithList = new List<Contact>()
+            { contact };
+
+            var returnedValue = DomainMapperWithContacts(contactWrappWithList);
+
+            await apiDbContext.Contacts!.AddRangeAsync(returnedValue);
+
+            await apiDbContext.SaveChangesAsync();
+
+            return returnedValue!.FirstOrDefault() !;
+        }
+
+        public async Task<Contact> UpdateContact(Contact contact)
+        {
+            var contactWrappWithList = new List<Contact>()
+            { contact };
+
+            var returnedValue = DomainMapperWithContacts(contactWrappWithList);
+
+            apiDbContext.Contacts!.UpdateRange(returnedValue);
+
+            await apiDbContext.SaveChangesAsync();
+
+            return contactWrappWithList!.FirstOrDefault() !;
+        }
+
+        public List<Contact> DomainMapperWithContacts(List<Contact> contacts)
+        {
+            var domainsQueryResult = (from ed in contacts
```

Comment:
@WASandaruwan very complicated code with copy-pasting of the same functionality (like you do Split("@") twice which is double complexity, especially with a big number of contacts). Please do not use words shortening.